_summaries/bottou2010large_2.md

```diff
@@ -1,19 +1,126 @@
+
+How Stochastic Gradient Decent (SGD) works is fairly similar to Gradient Decent (it's actually simpler)
+
+Given a training set pair $$x$$ and $$y$$, and a parameterized function $$f_{w}(x)$$ where $$w$$ is the weights of the parameterized function, 
+we can create a loss function $$l(f_{w}(x), y)$$ that compares how costly our prediction was from the correct output.  
+
+This loss function can be represented as $$Q(z_{i}, w_{t})$$ where $$z_{i}$$ is the ith training pair, and $$w_{t}$$ are the weights at timestep $$t$$.
+
+SGD says that you can update the weights of the function using a single training pair producing
+
+$$
+w_{t+1} = w_{t} - \gamma_{t} \nabla_{w}Q(z_{i},w_{t})
+$$
+
+effectively allowing us to optimize our weights a single randomly sampled training example at a time
+
+_**What are Risks?**_
+
+Ultimately, we want to minimize loss.
+
+Ideally this loss would be some measure of accuracy over the entire unknown distribution $$dP(z)$$ 
+
+If we could do this, we end up with _Expected Risk_
+
+$$
+E(f) = \int l(f(x), y)dP(z)
+$$
+
+However, we often do not know the entire distribution, we only have a subset of examples, our Training Set.
+
+Minimizing loss over a Training Set is the _Empirical Risk_
+
+$$
+E_{n}(f) = \frac{1}{n}\sum^{n}_{i=1}l(f(x_{i}), y_{i})
+$$
+
+The hope is that the Empirical Risk is a good approximation of the Expected Risk and thus reducing loss in the Empirical should give a reduction in loss on the Expected.
+
+_**How does Gradient Decent and Stochastic Gradient Decent work with Risks?**_
+
+Gradient Decent is able to reach the Empirical Optimum under sufficient assumptions on the learning rate, optimization, etc.
+
+SGD needs the same assumptions as Gradient Decent, but also needs decaying learning rates to cope with the noise in the gradient approximations.
+
+Because of these assumptions and the noise SGD makes, it can take longer for SGD to reach the Empirical Optimum.
+
+_**So... Why use SGD?**_
+
+When tasks are sufficiently large and computationally bound its usually better to get within some threshold of the Empirical Optimum
+
+$$
+E_{n}(\tilde{f}_{n}) < E_{n}(f_{n}) + p
+$$
+
+(where $$p$$ is the extra risk you are willing to accept to decrease training time)
+
+Now we have set a predefined accuracy that we want our algorithm to reach, the error between our predefined accuracy and the best possible accuracy we could ever get is called _Excess Error_
+
+$$
+\mathcal{E} = \mathbb{E} [E(\tilde{f}_{n}) - E(f^{*})]
+$$
+
+The authors found that SGD is quicker to reach a predefined Excess Error
+
+With sufficient training samples, you can model the Expected Risk (the actual distribution) more accurately, and be able to approach some predefined value of Expected Risk quicker with SGD than GD can.
+
+
+_**Does 2nd Order Stochastic Gradient Decent (2SGD) have any benefits?**_
+
+2SGD, with sufficient assumptions (convexity and regularity), will approach optimum solution as quickly as possible
```

Comment:
On page 2 of the paper, 2GD is brought up in slight detail. I think it might be valuable to discuss it here in a sentence or two to compare to 2SGD

Reply:
Totally, I forgot to add the equation for it and a quick explanation.  Good catch! I added the eq and very short sentence about how its different.